app/models/conversion_host.rb

```diff
@@ -4,4 +4,105 @@ class ConversionHost < ApplicationRecord
   acts_as_miq_taggable
 
   belongs_to :resource, :polymorphic => true
+
+  def active_tasks
+    ServiceTemplateTransformationPlanTask.where(:state => 'active').select do |task|
+      task.conversion_host == self
+    end
+  end
+
+  def eligible?
+    return true if concurrent_transformation_limit.nil?
+    active_tasks.size < concurrent_transformation_limit.to_i
+  end
+
+  def source_transport_method
+    return 'vddk' if vddk_transport_supported
+    return 'ssh' if ssh_transport_supported
+  end
+
+  def conversion_options(task)
```

Comment:
If you do change the task to a true relationship, then this method could probably be replaced with a number of delegate methods.

See [comment to create relationship](https://github.com/ManageIQ/manageiq/pull/18033/files#r221281524).

Reply:
Definitely. What release are we targeting there ? Hammer or I-release ? Given that we're past code freeze for Hammer, I'd say, we're aiming I-release, unless we've got a derogation :stuck_out_tongue_winking_eye:

Reply:
Well, the relationship is made so task belongs to conversion_host. Then, I'm not sure how I can delegate from conversion_host to task, as it has many tasks. That would require that the task is passed to the conversion. Is there a way to do it ? I'm a delegation newbie.